src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -69,12 +70,14 @@ private void updateExercises() {
     }
     Authentication auth = authentication.get();
     if (course == null || auth == null) {
-      ExercisesTreeViewModel emptyModel =
-          new ExercisesTreeViewModel(new ArrayList<>(), new Options());
+      ExercisesTreeViewModel emptyModel = new ExercisesTreeViewModel(new ArrayList<>(),
+              new Options());
       if (course == null) {
-        emptyModel.setEmptyTextVisible(true);
+        emptyModel = new EmptyExercisesTreeViewModel(new ArrayList<>(), new Options());
       }
```

Comment:
I think the inner `if (course == null)` could be removed and just directly assign `new EmptyExercise...` to `emptyModel`